app/main.py

```diff
@@ -2,0 +1,39 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+    @classmethod
+    def check_alive(cls) -> list:
+        cls.alive = [animal for animal in cls.alive if animal.health > 0]
+
+    @classmethod
+    def __str__(cls) -> str:
+        return str(cls.alive)
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, prey: "Herbivore") -> None:
+        if isinstance(prey, Herbivore) and not prey.hidden:
+            prey.health -= 50
+            Animal.check_alive()
```

Comment:
The `bite` method in `Carnivore` class calls `Animal.check_alive()` after reducing the prey's health. This is correct, but ensure that `check_alive` is intended to be called every time a bite occurs, as it will update the `alive` list immediately.